// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package fs;

service Api {
    rpc Execute(ExecuteRequest) returns (ExecuteResponse) {}
    rpc Hangup(HangupRequest) returns (HangupResponse) {}
    rpc Originate(OriginateRequest) returns (OriginateResponse) {}
}

message ErrorExecute {
    enum Type {
        ERROR = 0;
        USAGE = 1;
    }
    string message = 1;
    Type type = 2;
}

message ExecuteRequest {
    string command = 1;
    string args = 2;
}

message ExecuteResponse {
    string data = 1;
    ErrorExecute error = 2;
}

message HangupRequest {
    string uuid = 1;
    string cause = 2;
}

message HangupResponse {
    ErrorExecute error = 1;
}

message OriginateRequest {
    map<string, string> variables = 1;
    repeated string endpoints = 2;
    string destination = 3;
    int32 timeout = 4;
    string callerNumber = 5;
    string callerName = 6;
    string context = 7;
    string dialplan = 8;
}

message OriginateResponse {
    string uuid = 1;
    ErrorExecute error = 2;
}